File: 624-maximum-distance-in-arrays/maximum-distance-in-arrays.c
#define MAX(a, b) ((a) > (b) ? (a) : (b))
#define MIN(a, b) ((a) < (b) ? (a) : (b))
#define ABS(a) ((a) < 0 ? -(a) : (a))

int maxDistance(int** arrays, int arraysSize, int* arraysColSize) {
    int result = 0;
    int min = arrays[0][0];
    int max = arrays[0][arraysColSize[0] - 1];
    
    for (int i = 1; i < arraysSize; i++) {
        result = MAX(result, MAX(ABS(arrays[i][arraysColSize[i] - 1] - min),
                                 ABS(max - arrays[i][0])));
        min = MIN(min, arrays[i][0]);
        max = MAX(max, arrays[i][arraysColSize[i] - 1]);
    }
    
    return result;
}

File: 624-maximum-distance-in-arrays/maximum-distance-in-arrays.java
class Solution {
    public int maxDistance(List<List<Integer>> arrays) {
        int result = 0;
        int min = arrays.get(0).get(0);
        int max = arrays.get(0).get(arrays.get(0).size() - 1);
        
        for (int i = 1; i < arrays.size(); i++) {
            result = Math.max(result, Math.max(Math.abs(arrays.get(i).get(arrays.get(i).size() - 1) - min),
                                               Math.abs(max - arrays.get(i).get(0))));
            min = Math.min(min, arrays.get(i).get(0));
            max = Math.max(max, arrays.get(i).get(arrays.get(i).size() - 1));
        }
        
        return result;
    }
}